invite/app.ts

```diff
@@ -15,28 +15,109 @@
  */
 
 import {Application, Context} from 'probot';
+import {createTokenAuth} from '@octokit/auth';
+import Webhooks from '@octokit/webhooks';
+import {Octokit} from '@octokit/rest';
 
-module.exports = (app: Application) => {
+import {InviteBot} from './src/invite_bot';
+
+type WebhookHandler<T> = (inviteBot: InviteBot, payload: T ) => Promise<void>;
+
+type CommentWebhookPayload =
+  | Webhooks.WebhookPayloadIssues
+  | Webhooks.WebhookPayloadIssueComment
+  | Webhooks.WebhookPayloadPullRequest
+  | Webhooks.WebhookPayloadPullRequestReview
+  | Webhooks.WebhookPayloadPullRequestReviewComment;
+
+export default (app: Application) => {
   if (process.env.NODE_ENV !== 'test') {
     require('dotenv').config();
   }
 
-  app.on(
-    [
-      'issue_comment.created',
-      'issue.opened',
-      'pull_request.opened',
-      'pull_request_review.submitted',
-      'pull_request_review_comment.created',
-    ],
-    async (context: Context) => {
-      context.log.info(`Received ${context.event}.${context.payload.action}`);
-      context.log.info('TODO: Process the comment');
+  const github = new Octokit({
+    authStrategy: createTokenAuth,
+    auth: process.env.GITHUB_ACCESS_TOKEN,
+  });
+
+  /** Listens for webhooks and provides a bot instance to the handler. */
+  function listen(
```

Comment:
Not feeling super strongly about this, but I prefer you just have the `app.on` calls directly. Having `listen` doesn't add much for clarity and doesn't do much in way of de-duplication (instead of calling `handler()` 5 times, you call `listen()` 5 times...)

Reply:
Alrighty, donezo